exiftool_test.go

```diff
@@ -322,4 +362,114 @@ func TestSetExiftoolBinaryPath(t *testing.T) {
+	testCases := []struct{
+		name string
+		testResp string
+		expectErr bool
+	}{
+		{name: "token as full resp", testResp: writeMetadataSuccessToken, expectErr: false},
+		{name: "token at resp end", testResp: "prefix text" + writeMetadataSuccessToken,
+			expectErr: false},
+		{name: "token at resp middle",
+			testResp: "prefix text" + writeMetadataSuccessToken + "suffix text",
+			expectErr: true},
+		{name: "token at resp beginning", testResp: writeMetadataSuccessToken + "suffix text",
+			expectErr: true},
+		{name: "no token", testResp: "some error message", expectErr: true},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T){
+			err := handleWriteMetadataResponse(tc.testResp)
+			if tc.expectErr {
+				assert.Error(t, err)
+			} else {
+				assert.Nil(t, err)
+			}
+		})
+	}
+}
+
+func TestWriteMetadata(t *testing.T) {
+	t.Parallel()
+
+	fields := map[string]interface{} {
+		"title": "fake title",
+		"description": "fake description",
+	}
+
+	runWriteTest(t, func(t *testing.T, tmpDir string) {
+		e, err := NewExiftool()
+		require.Nil(t, err)
+
+		testCases := []struct{
+			md FileMetadata
+			expectErr bool
+		}{
+			{md: FileMetadata{File: filepath.Join(tmpDir, "20190404_131804.jpg"), Fields: fields},
+				expectErr: false},
+			{md: FileMetadata{File: filepath.Join(tmpDir, "binary.mp3"), Fields: fields},
+				expectErr: true},
+			{md: FileMetadata{File: filepath.Join(tmpDir, "empty.jpg"), Fields: fields},
+				expectErr: true},
+			{md: FileMetadata{File: filepath.Join(tmpDir, "extractEmbedded.mp4"), Fields: fields},
+				expectErr: false},
+		}
+
+		mds := make([]FileMetadata, 0, len(testCases))
+		for _, tc := range testCases {
+			mds = append(mds, tc.md)
+		}
+
+		e.WriteMetadata(mds)
+		for i, md := range mds {
+			if testCases[i].expectErr {
+				assert.Error(t, md.Err)
+			} else {
+				assert.Nil(t, md.Err)
+			}
+		}
+	})
+}
+
+func TestWriteMetadataInvalidField(t *testing.T) {
+	t.Parallel()
+
+	fields := map[string]interface{} {
+		"not a valid field": "invalid field value",
+	}
+
+	runWriteTest(t, func(t *testing.T, tmpDir string) {
+		e, err := NewExiftool()
+		require.Nil(t, err)
+
+		mds := []FileMetadata{
+			{File: filepath.Join(tmpDir, "20190404_131804.jpg"), Fields: fields},
+			{File: filepath.Join(tmpDir, "binary.mp3"), Fields: fields},
+			{File: filepath.Join(tmpDir, "empty.jpg"), Fields: fields},
+			{File: filepath.Join(tmpDir, "extractEmbedded.mp4"), Fields: fields},
+		}
+
+		e.WriteMetadata(mds)
+		for _, md := range mds {
+			assert.Error(t, md.Err)
+		}
+	})
+}
+
+func runWriteTest(t *testing.T, f func(t *testing.T, tmpDir string)) {
+	tmpDir, err := os.MkdirTemp("", "testdata*")
+	require.Nil(t, err, "Unable to create temporary directory")
+	defer func() {
+		err := os.RemoveAll(tmpDir)
+		assert.Nil(t, err, "Unable to remove temporary directory: " + tmpDir)
+	}()
+
+	err = copy.Copy("testdata", tmpDir, copy.Options{Sync: true});
```

Comment:
I'd prefer to limit as much as possible external dependencies to avoid hypothetical conflicts with the client. Could you please copy the test files in tmpDir using the standard library ? 

Here is a way to do it (https://github.com/barasher/picdexer/blob/master/internal/filewatcher/fileWatcher_test.go):
```golang
func copy(src, dest string) error {
	b, err := os.ReadFile(src)
	if err != nil {
		return err
	}
	err = os.WriteFile(dest, b, 0644)
	if err != nil {
		return err
	}
	return nil
}
```
And I realize that I could be more concise with this function :)

Reply:
Ahh, I wanted to avoid implementing copy but will go ahead and do this. Pulling in an dependency isn't ideal either...